phoenix-core/src/main/java/org/apache/phoenix/mapreduce/OrphanViewTool.java

```diff
@@ -0,0 +1,812 @@
+            throws Exception {
+        ResultSet physicalLinkRS = phoenixConnection.createStatement().executeQuery(physicalLinkQuery);
+        while (physicalLinkRS.next()) {
+            String tenantId = physicalLinkRS.getString(1);
+            String schemaName = physicalLinkRS.getString(2);
+            String tableName = physicalLinkRS.getString(3);
+            Key viewKey = new Key(tenantId, schemaName, tableName);
+            View view = orphanViewSet.get(viewKey);
+
+            String baseTenantId = physicalLinkRS.getString(4);
+            String baseFullTableName = physicalLinkRS.getString(5);
+            Key baseKey = new Key(baseTenantId, baseFullTableName);
+            Base base = baseSet.get(baseKey);
+
+            if (view == null || base == null) {
+                orphanLinkSet.add(new Link(viewKey, baseKey, PTable.LinkType.PHYSICAL_TABLE));
+            }
+            else {
+                view.setBase(baseKey);
+            }
+        }
+    }
+
+    /**
+     * Go through all the child-parent links in the system child-link table and update the parent field of the
+     * view objects of orphanViewSet. Check if the child does not exist add the link to orphanLinkSet.
+     * @param phoenixConnection
+     * @throws Exception
+     */
+    private void processChildParentLinks(PhoenixConnection phoenixConnection)
+            throws Exception {
+        ResultSet childParentLinkRS = phoenixConnection.createStatement().executeQuery(childParentLinkQuery);
+        while (childParentLinkRS.next()) {
+            String childTenantId = childParentLinkRS.getString(1);
+            String childSchemaName = childParentLinkRS.getString(2);
+            String childTableName = childParentLinkRS.getString(3);
+            Key childKey = new Key(childTenantId, childSchemaName, childTableName);
+            View childView = orphanViewSet.get(childKey);
+
+            String parentTenantId = childParentLinkRS.getString(4);
+            String parentFullTableName = childParentLinkRS.getString(5);
+            Key parentKey = new Key(parentTenantId, parentFullTableName);
+            View parentView = orphanViewSet.get(parentKey);
+
+            // Check if parentTenantId is not set but it is the same as the childTenantId. Is this a bug?
+            if (parentView == null && parentTenantId == null && childTenantId != null) {
+                parentKey = new Key(childTenantId, parentFullTableName);
+                parentView = orphanViewSet.get(parentKey);
+            }
+
+            if (childView == null || parentView == null) {
+                orphanLinkSet.add(new Link(childKey, parentKey, PTable.LinkType.PARENT_TABLE));
+            }
+            else {
+                childView.setParent(parentKey);
+            }
+        }
+    }
+
+    /**
+     * Go through all the parent-child links in the system catalog table and update the parent field of the
```

Comment:
Nit: Comment should say "system child-link table" instead of system catalog, no?

Reply:
I will correct the comments